streams/src/test/java/org/apache/kafka/streams/integration/AbstractResetIntegrationTest.java

```diff
@@ -167,11 +168,24 @@ public boolean conditionMet() {
     private final static Properties RESULT_CONSUMER_CONFIG = new Properties();
 
     void prepareTest() throws Exception {
-        cluster.deleteAndRecreateTopics(INPUT_TOPIC, OUTPUT_TOPIC, OUTPUT_TOPIC_2, OUTPUT_TOPIC_2_RERUN);
-
         prepareConfigs();
         prepareEnvironment();
 
+        // busy wait until cluster (ie, ConsumerGroupCoordinator) is available
+        while (true) {
```

Comment:
Are we sure we want `while(true)`? Should we not give up after some period of time so the build doesn't hang?

Also, why can't we just put a longer wait in the `TestUtils.waitForCondition(...)`?

Reply:
The `while(true)` was in the code before the rewrite an not an issue.

A long wait won't do as `TestUtils.waitForCondition(...)` throws an `AsssertionException` if its wait time passed, while we get a `TimeoutException` from the request itself here if the broker is not available yet. I guess, we could increase the timeout for the request, but we reuse the same condition on other places and their we don't want to have  timeout -- could parametrize it of course....

Let me know what you think.